src/com/amazon/ion/impl/bin/utf8/Utf8StringEncoderPool.java

```diff
@@ -0,0 +1,60 @@
+package com.amazon.ion.impl.bin.utf8;
+
+import java.util.concurrent.ArrayBlockingQueue;
+
+/**
+ * A thread-safe shared pool of {@link Utf8StringEncoder}s that can be used for UTF8 encoding and decoding.
+ */
+public class Utf8StringEncoderPool {
+    // The maximum number of Utf8Encoders that can be waiting in the queue before new ones will be discarded.
+    private static final int MAX_QUEUE_SIZE = 32;
```

Comment:
Out of curiosity, was this just a small number that seemed reasonable or did you get this number from something?

Reply:
Just a number that seemed reasonable. In the worst case, an application that had more than 32 binary writers in existence at the same time would be allocating fresh `Utf8StringEncoder`s for the surplus, which seemed low stakes. 

That said, raising the ceiling is pretty cheap, though. Each `Utf8StringEncoder` is something like ~20KB on the heap and the queue only allocates them as needed, so I might bump this up to 128 while I'm making tweaks.